### PR TITLE
toggle for stripping empty shell/command output

### LINKED ISSUE #45367: Empty lines are stripped from the command module's output.
##### SUMMARY
The command module strips trailing newlines from the output.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
command

##### ANSIBLE VERSION
```paste below
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/merrily/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION
```paste below

```

##### STEPS TO REPRODUCE
Give the command module a program that outputs blank lines.

```yaml
#playbook.yml
---
- hosts: 127.0.0.1
  tasks:
    - shell: echo -e "\n\n\n"
      register: result

    - debug:
        var: result.stdout
```

##### EXPECTED RESULTS
I expect to see the exact program output.

##### ACTUAL RESULTS
I did not get to see the exact program output.

```paste below
 [WARNING]: provided hosts list is empty, only localhost is available. Note
that the implicit localhost does not match 'all'


PLAY [127.0.0.1] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [127.0.0.1]

TASK [shell] *******************************************************************
changed: [127.0.0.1]

TASK [debug] *******************************************************************
ok: [127.0.0.1] => {
    "result.stdout": ""
}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=3    changed=1    unreachable=0    failed=0   


```

### PR DESCRIPTION
**Pull request by bcoca**
fixes #45367

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

shell/command

### CODE REVIEW

**Comment by angystardust** [comment]
Hi @bcoca, sorry for the noise...
Is it possible to add and option to the _command_ module to control stripping of empty chars at begin/end of the output? An example use-case would be the use of lvm **lvs** command which gives some extra spaces at the begin of the output and so one need to use the shell module piping the output with _sed_ (and that's a little bit sad :) )
```
        - name: Get the root volume-group name
          shell: lvs --noheadings -o vg_name {{ devicemapper_devname }} | sed 's/^ *//'
          changed_when: false
          register: vgname
```
Would be great to use the command module instead:
```
        - name: Get the root volume-group name
          command: lvs --noheadings -o vg_name {{ devicemapper_devname }}
          strip_empty_chars: yes
          changed_when: false
          register: vgname
```

Does it make sense at all? Should I open a new feature request for that?
P.s.: I see here you're talking about stripping lines and not about chars but strip is still the topic :)

File: lib/ansible/modules/commands/command.py [hunk 1]
```diff
@@ -69,6 +69,12 @@
     description:
       - If set to C(yes), append a newline to stdin data.
     version_added: "2.8"
+  strip_empty_ends:
+    description:
+      - Strim empty lines from the end of stdout/stderr in result.
```
Comment (Akasurde) [inline lib/ansible/modules/commands/command.py]:
```suggestion
      - Strip empty lines from the end of stdout/stderr in result.
```

**Comment by bcoca** [comment]
@angystardust I would just use Jinja2 to handle that, the reason I'm adding this option is because some users needed the content we were automatically striping already and we need to keep backwards compatibility, otherwise i would have just removed the stripping and let users trim as they see fit in templating.

**Comment by jsosic** [comment]
Thank you very much! <3

**Pull request merged**